openedx/core/lib/api/views.py

```diff
@@ -0,0 +1,25 @@
+from rest_framework import generics
+from rest_framework.views import APIView
+from openedx.core.lib.api.permissions import ApiKeyHeaderPermission
+from openedx.core.lib.api.serializers import PaginationSerializer
+
+
+class PermissionMixin(object):
```

Comment:
This should have a more specific name; we don't expect all (or even most) of our APIs to do auth in this way.